### PR TITLE
WIP: Removed CoreAppHost implementation of ApplicationHost

### PR DESCRIPTION
**Pull request by wtayl0r**
CoreAppHost didn't provide anything over ApplicationHost and as we'll
only have one implementation of ApplicationHost, it makes sense to
remove abstract and add CoreAppHost's functionality to ApplicationHost.

As a side effect of this, SocketSharp had to be moved to
Server.Implementations which is where it should have been anyway.

Marked WIP to get feedback on the removal of CoreAppHost and to make sure GetEntryAssembly gives us the same result as typeof(CoreAppHost).Assembly.

### CODE REVIEW

**Comment by cvium** [comment]
Makes sense to me. Maybe the point of the abstract class is to have separate application host for Nvidia Shield TV (Android TV) etc?

File: Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs [hunk 1]
```diff
@@ -326,7 +326,7 @@ public static string LeftPart(string strVal, char needle)
                 return null;
             }
 
-            var pos = strVal.IndexOf(needle, StringComparison.Ordinal);
+            var pos = strVal.IndexOf(needle.ToString(), StringComparison.Ordinal);
```
Comment (Bond-009) [inline Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs]: Why this change?

File: Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs [hunk 2]
```diff
@@ -307,11 +307,11 @@ private static string GetQueryStringContentType(IRequest httpReq)
             }
 
             format = LeftPart(format, '.');
-            if (format.Contains("json", StringComparison.OrdinalIgnoreCase))
+            if (format.ToLower().Contains("json"))
```
Comment (Bond-009) [inline Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs]: Don't use `ToLower` use the overload with `StringComparison.OrdinalIgnoreCase`

File: Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs [hunk 3]
```diff
@@ -42,7 +42,7 @@ private static string GetHandlerPathIfAny(string listenerUrl)
             }
 
             var startHostUrl = listenerUrl.Substring(pos + "://".Length);
-            var endPos = startHostUrl.IndexOf('/', StringComparison.Ordinal);
+            var endPos = startHostUrl.IndexOf("/", StringComparison.Ordinal);
```
Comment (Bond-009) [inline Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs]: Why change this?

Comment (EraYaN) [reply Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs]: IndexOf(char) might also be faster, then the string one.

File: Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs (hunk 1 above)
Comment (wtayl0r) [reply Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs]: Apologies for the late reply, dotnet standard doesn't seem to have the implementation. The code used to be in Jellyfin.Server which is a dotnet core project which does have this overload. The code is now in Server.Implementations, a dotnet standard project, which is why I had to make this change.

File: Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs (hunk 2 above)
Comment (wtayl0r) [reply Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs]: As above

Comment (Bond-009) [reply Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs]: Then use `.IndexOf` instead of `.Contains`

**Comment by JustAMan** [comment]
@wtayl0r your PR now has merge conflicts, could you please fix them?

**Comment by cvium** [comment]
FYI, this PR conflicts with #1010. Could we hold off on this one and rebase it on top of #1010 when it's ready?

**Comment by wtayl0r** [comment]
> FYI, this PR conflicts with #1010. Could we hold off on this one and rebase it on top of #1010 when it's ready?

100% agree, current changes make rebasing off of current master impossible, particularly around ReadOnlySpan<char> casting to string

**Comment by dkanada** [comment]
This has been sitting for a while but it does look like these classes might be important if we ever want to add support for different frameworks. Shield builds would require an Xamarin implementation for the entry point. @wtayl0r what do you think now that #1010 has been merged to master?

**Comment by wtayl0r** [comment]
Yes, I agree we should close this now due to #1010. This was a prerequisite to some dependency injection changes that I had/have planned, but it'd be easier now to start fresh off of master.
